tabular/src/autogluon/tabular/models/tabular_nn/tabular_nn_dataset.py

```diff
@@ -3,6 +3,7 @@
 import numpy as np
 import pandas as pd
 import mxnet as mx
+import torch
```

Comment:
Note that `torch` is an optional AG dependency at the moment. That means you should only import it where needed inside functions or at the top of files that are only imported in specific files. Otherwise AG will crash as soon as it tries to import torch. 
We need to make sure `try_import_torch()` is always called first at runtime (in `_fit()`) before any `import torch` lines are called.

Reply:
In fact, you should try testing running standard autogluon examples (not quantile-regression) in a fresh virtualenv, where you don't have torch installed. This should still work fine. 
And then also try running the quantile-regression without torch installed. This should produce an informative error message if `try_import_torch()` was properly invoked before any of the `import torch` calls.

Reply:
moving to new file `tabuler_nn_torch.py`

Reply:
I'm double checking this to avoid any serious error.

Reply:
After cleaning codes and properly locating `try_import_torch()`, I've got the following message by removing `pytorch`:
```	
Warning: Exception caused QuantileNeuralNet to fail during training (ImportError)... Skipping this model.
		Unable to import dependency torch
A quick tip is to install via `pip install torch`.
```
It seems like it is properly skipping the model without any disruption.